include/bluetooth/gatt.h

```diff
@@ -1441,6 +1441,24 @@ struct bt_gatt_subscribe_params {
 int bt_gatt_subscribe(struct bt_conn *conn,
 		      struct bt_gatt_subscribe_params *params);
 
+/** @brief Add Attribute Value Notification subscription
+ *
+ *  Add Attribute Value Notification when already subscribed from a previous
+ *  connection. The GATT server will remember subscription from previous
+ *  connections when bonded, so the subscription can be added without performing
+ *  a new subscribe procedure after a power cycle.
+ *
+ *  @note Notifications are asynchronous therefore the parameters need to
+ *        remain valid while subscribed.
+ *
+ *  @param id    Connection object.
+ *  @param params Subscribe parameters.
+ *
+ *  @return 0 in case of success or negative value in case of error.
+ */
+int bt_gatt_add_subscription(uint8_t id, const bt_addr_le_t *addr,
```

Comment:
Perhaps resubscribe is a better name here, now I do recall commenting on https://github.com/zephyrproject-rtos/zephyr/issues/21798#issuecomment-578267127, the idea would be tha we do restore from settings and put in a temporary area then when the application attempts to subscribe we check that area and if there is a subscription for the peer/handle/value don't write to the CCC.

Reply:
If the application is to subscribe without connecting and discovering the database then the application will have to remember the CCC handle and value handle, in addition to information needed to provide the correct subscribe params.
So in this case storing the handles in the stack would be duplicating storage.

I'm going with this approach to allow the application to resubscribe based on it's stored information.

Reply:
Fair enough, until we actually have proper caching as client I guess that is the best we can have, that said there have been some request to support general discovery with caching and in that case we could implement the resubscription along with the caching logic so the stack would remember what it had subscribed, that way the application don't need to store anything it just reload from cache.